### PR TITLE
Fix #302

### LINKED ISSUE #302: heap-buffer-overflow (out-of-bound-read) in yylex_MACROARGS
Hello.

I think, The 749 lines in lexer.c can fall into an infinite loop,  In case of abnormal file.

If the pLexBuffer does not have ("," , "\n") cannot exit the while statement.

what do you think?

Thanks.

lexer.c:749
```
while ((*pLexBuffer != ',') && (*pLexBuffer != '\n'))
```

https://github.com/rednex/rgbds/blob/7149fc1e397917824f1d3e06c7c281def7ec6b82/src/asm/lexer.c#L741-L844

### PR DESCRIPTION
**Pull request by dbrotz**
The line continuation character `\` is causing the newline that is automatically inserted at the end of the file to be eaten. Inserting two newlines fixes the issue and doesn't impact the lexer's performance.

Fixes #302.

### CODE REVIEW

**Comment by AntonioND** [comment]
The code looks fine, but the commit message should be descriptive and not refer to a GitHub issue. If the project were to move away from here the message wouldn't be useful anymore.

**Comment by dbrotz** [comment]
That makes sense. I've updated it.

**Comment by dbrotz** [comment]
I've updated it to only add the newlines if they are necessary. This ensures that nTotalLines has the correct value.

File: src/asm/lexer.c [hunk 1]
```diff
@@ -219,6 +216,30 @@ YY_BUFFER_STATE yy_create_buffer(FILE *f)
...
+	/* Add newline if \ will eat the last newline */
+	if (pBuffer->nBufferSize >= 2) {
+		size_t pos = pBuffer->nBufferSize - 2;
+
+		/* Skip spaces */
+		while (pos > 0 && pBuffer->pBuffer[pos] == ' ')
+			pos--;
+
+		if (pBuffer->pBuffer[pos] == '\\') {
+			pBuffer->pBuffer[pBuffer->nBufferSize] = '\n';
```
Comment (AntonioND) [inline src/asm/lexer.c]: So you allocate ``pBuffer->nBufferSize`` bytes, then write to ``pBuffer->pBuffer[pBuffer->nBufferSize]`` and "increase" the size by incrementing ``pBuffer->nBufferSize``, and then write 0 to the new end, which isn't allocated. This is wrong.

Comment (dbrotz) [reply src/asm/lexer.c]: I've changed the code a bit to make what it more obvious what's happening, but the behavior has not been changed.

**Comment by dbrotz** [comment]
I've changed the code a bit to make it more obvious what's happening, although the behavior has not been changed.

**Pull request merged**